analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -71,23 +72,20 @@ private static void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
                 return;
             }
 
-            var parameterNameValue = new Optional<object>();
-            var messageValue = new Optional<object>();
+            var parameterNameValue = default(Optional<object>);
+            var messageValue = default(Optional<object>);
             for (var i = 0; i < methodSymbol.Parameters.Length; i++)
             {
+                var argument = objectCreation.ArgumentList.Arguments[i];
                 var argumentExpression = objectCreation.ArgumentList.Arguments[i].Expression;
-                if (methodSymbol.Parameters[i].MetadataName == "paramName" || methodSymbol.Parameters[i].MetadataName == "parameterName")
-                {
-                    parameterNameValue = analysisContext.SemanticModel.GetConstantValue(argumentExpression);
-                }
-                else if (methodSymbol.Parameters[i].MetadataName == "message")
-                {
-                    messageValue = analysisContext.SemanticModel.GetConstantValue(argumentExpression);
-                }
-                else
-                {
-                    // do nothing
-                }
+                RetrieveParameterAndMessageArgumentValue(
+                    argument.NameColon != null
```

Comment:
IMO it would be more readable to extract the result of this expression into a nicely named variable